sharding-core/sharding-core-preprocessor/src/test/java/org/apache/shardingsphere/core/preprocessor/segment/select/projection/engine/ProjectionsContextEngineTest.java

```diff
@@ -59,4 +62,40 @@ public void assertProjectionsContextCreatedProperlyWhenSelectItemPresent() {
         ProjectionsContext projectionsContext = new ProjectionsContextEngine(null).createProjectionsContext(null, selectStatement, mock(GroupByContext.class), mock(OrderByContext.class));
         assertNotNull(projectionsContext);
     }
+    
+    @Test
+    public void createProjectionsContextWhenOrderByContextOrderItemsPresent() {
+        SelectStatement selectStatement = mock(SelectStatement.class);
+        SelectItemsSegment selectItemsSegment = mock(SelectItemsSegment.class);
+        when(selectStatement.getSelectItems()).thenReturn(selectItemsSegment);
+        ShorthandSelectItemSegment shorthandSelectItemSegment = new ShorthandSelectItemSegment(0, 10, "text");
+        TableSegment owner = new TableSegment(0, 10, "name");
+        owner.setOwner(new SchemaSegment(0, 10, "name"));
+        shorthandSelectItemSegment.setOwner(owner);
+        when(selectItemsSegment.getSelectItems()).thenReturn(Collections.<SelectItemSegment>singleton(shorthandSelectItemSegment));
+        OrderByContext orderByContext = mock(OrderByContext.class);
+        OrderByItem orderByItem = mock(OrderByItem.class);
+        when(orderByItem.getSegment()).thenReturn(mock(IndexOrderByItemSegment.class));
+        when(orderByContext.getItems()).thenReturn(Collections.singletonList(orderByItem));
+        ProjectionsContext actual = new ProjectionsContextEngine(null).createProjectionsContext(null, selectStatement, mock(GroupByContext.class), orderByContext);
+        assertNotNull(actual);
+    }
+    
+    @Test
+    public void assertProjectionsContextCreatedProperlyWhenOrderByItemSegmentNotInstanceOfIndexOrderByItemSegment() {
```

Comment:
Why the method name as `assertProjectionsContextCreated...`?
The method to be tested name is `createProjectionsContext`, the test method should keep consist with be tested one.

Reply:
the method name of assertProjectionsContextCreatedProperlyWhenOrderByItemSegmentNotInstanceOfIndexOrderByItemSegment express the meaning: assert that ProjectionsContext instance is created when OrderByItemSegment is not instance of InstanceOfIndexOrderByItemSegment, corresponding to the following switch in the format of negation:

![image](https://user-images.githubusercontent.com/6405801/66926479-b2467780-f060-11e9-9697-3be07bf8722d.png)


So, I think the method name of assertProjectionsContextCreatedProperlyWhenOrderByItemSegmentNotInstanceOfIndexOrderByItemSegment could not be changed.

Reply:
I don't think so. In the unit test conduct, the testing method's prefix should name as the to be tested method which to let user know what is actual testing method of this case.
The proposal of the test case is that test the method `createProjectionsContext`, so the test case name should be start with `assertCreateProjectionsContext`.

`assertCreateProjectionsContextWithoutIndexOrderByItemSegment` maybe a better name.

Reply:
OK